### PR TITLE
Add Reason of Absence to Attendance

### LINKED ISSUE #9: As a user, I want to indicate reason of absence for each student
So that I know whether the student has a valid reason for being absent that particular lesson.

### PR DESCRIPTION
**Pull request by anthonytamzil**
Addresses #9 

Changes:
- Add Week class 
- Refactor to include Week arguments in Attendance constructor instead of LocalDate so week number has to be included while taking attendance with `w/`
- Users are now required to add a reason of absence with `r/` for students who are marked as absent
- Reason of absence is not needed for students who are marked as present
- Add new detailed message when attendance is updated or marked
- Standardized prefixes for `MarkAttendanceCommand` and `FindCommand`
- Updated user guide 
- Enabled assertions in `build.gradle`
- Added test cases for ParseWeek, Week, ParseID

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java [hunk 1]
```diff
@@ -89,15 +98,20 @@ public void parse_unexpectedPreamble_throwsParseException() {
...
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkAttendanceCommand.MESSAGE_USAGE));
 
         // Missing attendance prefix
-        assertParseFailure(parser, " /name " + VALID_NAME_AMY + " 1",
+        assertParseFailure(parser, " " + PREFIX_NAME + " " + VALID_NAME_AMY + " 1 " + PREFIX_WEEK + " 1",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkAttendanceCommand.MESSAGE_USAGE));
+
+        // Missing week prefix
+        assertParseFailure(parser, " " + PREFIX_NAME + " " + VALID_NAME_AMY + " 1",
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkAttendanceCommand.MESSAGE_USAGE));
```
Comment (spatuly) [inline src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java:115]: Can consider adding a test case for missing reason prefix when attendance is 0

File: src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java [hunk 2]
```diff
@@ -75,11 +82,13 @@ public void parse_validArgsWithMultipleIds_returnsMarkAttendanceCommand() {
...
         // Unexpected preamble data "unexpected" before the valid arguments
-        assertParseFailure(parser, "unexpected /name " + VALID_NAME_AMY + " /attendance 1",
+        assertParseFailure(parser, "unexpected " + PREFIX_NAME + " " + VALID_NAME_AMY + " "
+                        + PREFIX_ATTENDANCE + " 1",
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkAttendanceCommand.MESSAGE_USAGE));
 
         // Unexpected preamble data "unexpected" before the valid arguments with ID
-        assertParseFailure(parser, "unexpected /id " + VALID_ID_AMY + " /attendance 0",
+        assertParseFailure(parser, "unexpected " + PREFIX_ID + " " + VALID_ID_AMY + " "
+                        + PREFIX_ATTENDANCE + " 0",
```
Comment (spatuly) [inline src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java:91]: Would this throw a parse exception without the "unexpected" as attendance is 0 but no reason is given?

File: src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java [hunk 3]
```diff
@@ -59,20 +90,40 @@ public CommandResult execute(Model model) throws CommandException {
...
+                    successMessage.append(String.format(MESSAGE_SUCCESS + "%s\n%s" + MESSAGE_PRESENT + "%d\n",
+                            targetPerson.getName(), targetPerson.getName(), attendance.getWeek().getWeekNumber()));
+                } else {
+                    successMessage.append(String.format(MESSAGE_SUCCESS + "%s\n%s" + MESSAGE_ABSENT
+                                    + "%d\nReason: %s\n",
+                            targetPerson.getName(), targetPerson.getName(), attendance.getWeek().getWeekNumber(),
+                            attendance.getReason()));
+                }
             }
         }
```
Comment (spatuly) [inline src/main/java/seedu/address/logic/commands/MarkAttendanceCommand.java:129]: Not very important right now but this method might be too long?

**Review by spatuly — COMMENTED** [review]
LGTM!

File: src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java (hunk 1 above)
Comment (anthonytamzil) [reply src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java:115]: Yep! Will do.

File: src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java (hunk 2 above)
Comment (anthonytamzil) [reply src/test/java/seedu/address/logic/parser/MarkAttendanceCommandParserTest.java:91]: That's a valid point! Will tweak it a bit!

**Pull request merged**